datadog_checks_base/datadog_checks/base/stubs/similar.py

```diff
@@ -9,6 +9,66 @@
 MAX_SIMILAR_TO_DISPLAY = 15
 
 
+def dict_diff(expected, closest):
+    """
+    Returns an array of key/value pairs that are different between the two dicts.
+    The number of keys is fairly small so we can prioritize clarity over performance.
+    """
+    diff = []
+    for key in closest:
+        expected_value = expected.get(key)
+        closest_value = closest.get(key)
+
+        if expected_value is not None and expected_value != closest_value:
+            diff.append((key, expected_value, closest_value))
+    for key in expected:
+        if expected[key] is not None and key not in closest:
+            diff.append((key, expected[key], None))
+
+    return diff
+
+
+def tags_list_to_dict(tags):
+    return {tag.split(':', 1)[0]: (tag.split(':', 1)[1] if ":" in tag else '') for tag in tags}
+
+
+def tags_diff(expected, closest):
+    """
+    Returns an array of key/value pairs that are different between the two lists of tags.
+    """
+    diff = []
+    expected_tags_dict = tags_list_to_dict(expected)
+    closest_tags_dict = tags_list_to_dict(closest)
+    for tag in expected_tags_dict:
+        if expected_tags_dict[tag] != closest_tags_dict.get(tag):
+            diff.append((tag, expected_tags_dict[tag], closest_tags_dict.get(tag)))
+    for tag in closest_tags_dict:
+        if tag not in expected_tags_dict:
+            diff.append((tag, None, closest_tags_dict[tag]))
+    return diff
+
+
+def metric_stub_diff(expected, closest):
```

Comment:
I would include something about `format` in the name of this function. Otherwise I mentally group it with the other 2 `..._diff` functions, but those don't care about formatting.